File: BaekJoon/python/10828_stack.py
import sys


class MyStack:
    def __init__(self):
        self.stack = []
    
    def push(self, item):
        self.stack.append(item)
    
    def pop(self):
        if self.is_empty():
            return -1
        return self.stack.pop()
    
    def top(self):
        if self.is_empty():
            return -1
        return self.stack[-1]
    
    def is_empty(self):
        return int(len(self.stack) == 0)
    
    def size(self):
        return len(self.stack)


if __name__ == "__main__":
    n = int(input())
    stack = MyStack()
    
    for _ in range(n):
        command = sys.stdin.readline().split()
        status = command[0]
        
        if status == "push":
            stack.push(command[1])
        elif status == "pop":
            print(stack.pop())
        elif status == "top":
            print(stack.top())
        elif status == "empty":
            print(stack.is_empty())
        elif status == "size":
            print(stack.size())

File: BaekJoon/cpp/10828_stack.cpp
#include <iostream>
#include <vector>

struct MyStack {
    std::vector<int> arr;

    MyStack() = default;
    int top();
    void push(int data);
    int pop();
    [[nodiscard]] bool empty() const;
    [[nodiscard]] size_t size() const;
};

int MyStack::top() {
    if (arr.empty())
        return -1;
    return arr.back();
}

void MyStack::push(int data) {
    arr.push_back(data);
}

int MyStack::pop() {
    if (arr.empty())
        return -1;
    int last = arr.back();
    arr.pop_back();
    return last;
}

bool MyStack::empty() const {
    return arr.empty();
}

size_t MyStack::size() const {
    return arr.size();
}

int main() {
    int n;
    std::cin >> n;

    auto * stack = new MyStack();

    for (int iter = 0; iter < n; ++iter) {
        std::string status;
        int data;

        std::cin >> status;
        if (status == "top")
            std::cout << stack->top() << std::endl;
        else if (status == "push") {
            std::cin >> data;
            stack->push(data);
        }
        else if (status == "pop")
          std::cout << stack->pop() << std::endl;
        else if (status == "empty")
          std::cout << stack->empty() << std::endl;
        else if (status == "size")
          std::cout << stack->size() << std::endl;
    }

    return 0;
}
